packages/flutter/lib/src/cupertino/activity_indicator.dart

```diff
@@ -120,28 +152,30 @@ class _CupertinoActivityIndicatorPainter extends CustomPainter {
   final Animation<double> position;
   final RRect tickFundamentalRRect;
   final Color activeColor;
+  final double radius;
+  final double progress;
 
   @override
   void paint(Canvas canvas, Size size) {
     final Paint paint = Paint();
 
     canvas.save();
-    canvas.translate(size.width / 2.0, size.height / 2.0);
+    canvas.translate(size.width / 2.0, radius);
```

Comment:
This moves the center of the spinner to the top center from the center of the canvas, if the canvas has a big enough size. Is it because the activity indicator needs to be pinned to the top when it's in `done` state?

Reply:
Yes, precisely. Once the drag is fully complete, `size.height / 2.0` should be the same as `radius`, however where we were coming unstuck was that during the initial stages of the drag, `size.height` is less than the diameter of the spinner and changes as the user drags. This has the undesirable effect of moving the spinner as the user drags down.

By using radius, we ensure that the translation is constant and the spinner doesn't move. I've added some comments to this effect.

Reply:
Activity indicator is a public widget so we probably shouldn't change its behavior just to suit our needs. I think we might want to change the `SliverGeometry` the `_RenderCupertinoSliverRefresh` reports to its viewport  (or we can add it to the list of things we want to improve later).

Reply:
Ah, of course ... that makes total sense. I've reverted that change and pushed a change that manipulates the sliver geometry. 

While I was in there, I realised that the standard iOS pull-to-refresh interaction has a super-subtle effect where the first few pixels of the top tick get drawn out of the clipping bounds over the top of the content being dragged down. By adding the margin to the `paintOrigin` I achieve the same effect... however, it has the knock-on effect of impacting a couple of testcases.

At the time those test cases were written, the widget bounds that it was checking were correct, but I've had to adjust them slightly to accommodate these sliver geometry changes.

Let me know if I'm on the wrong path, or if there's a better way.